qiskit_machine_learning/algorithms/trainable_model.py

```diff
@@ -141,9 +142,42 @@ def initial_point(self, initial_point: np.ndarray) -> None:
         """Sets the initial point"""
         self._initial_point = initial_point
 
+    @property
+    def weights(self) -> np.ndarray:
+        """
+        Returns:
+            an array of trained weights. This property is for convenience only, it does the same
+            as `model.fit_result.x`.
+
+        Raises:
+            QiskitMachineLearningError: If the model has not been fit.
+        """
+        self._is_fit()
+        return np.asarray(self._fit_result.x)
```

Comment:
"This property is for convenience only, it does the same as `model.fit_result.x`." I guess I have to as why not here then why not simply do `return self.fit_result.x` since that would do the fit check and avoid it being replicated here. Maybe you want to do the np.asarray so ``return np.asarray(self.fit_result.x)`  but then I'd argue " it does the same as `model.fit_result.x`."" is not exactly right -   since doing the np,asarray can change things given .x as per OptimizerResult, is a `float | np.ndarray`

Reply:
Rephrased, please check it out if the text is clear enough.

Reply:
Seems fine - as I mentioned this could have been just one line `return np.asarray(self.fit_result.x)` since going via the property would do a fit check avoiding this needing to be done here too. But I know you prefer to access the instance vars more directly, so I guess you prefer it as it is. Tha'ts fine too.